odb_extractor_atoms.py

```diff
@@ -59,98 +223,297 @@ def select_obsclass(classes):
+            if inst in ["'Alopeke", "Zorro"]:
+                fpu = 'None'
+            else:
+                fpu = inst
+        else:
+            fpu = step[fpuinst[inst]]
+        config['inst'] = inst
+        config['fpu'].append(fpu)
+
+        if 'instrument:disperser' in step_keys:
+            disperser = step['instrument:disperser']
+        elif inst in ['IGRINS', 'MAROON-X']:
+            disperser = inst
+        else:
+            disperser = 'None'
+        if inst == 'GNIRS':
+            if step['instrument:acquisitionMirror'] == 'in' and \
+                    step['instrument:decker'] == 'acquisition':
+                disperser = 'mirror'
+            else:
+                disperser = disperser.strip('grating') + step['instrument:crossDispersed']
+        elif inst == 'Flamingos2' and fpu == 'FPU_NONE':
+            if step['instrument:decker'] == 'IMAGING':
+                disperser = step['instrument:decker']
+        config['disperser'].append(disperser)
+
+        if 'instrument:filter' in step_keys:
+            filter = step['instrument:filter']
+        elif inst == 'GPI':
+            filter = find_filter(fpu, gpi_filter_wav)
+        else:
+            if inst == 'GNIRS':
+                filter = 'None'
+            else:
+                filter = 'Unknown'
+        if inst == 'NIFS' and 'Same as Disperser' in filter:
+            for filt in list(nifs_filter_wav.keys()):
+                if disperser[0] in filt:
+                    filter = filt
+                    break
+        config['filter'].append(filter)
+
+        if inst == 'GPI':
+            wavelength = gpi_filter_wav[filter]
+        else:
+            wavelength = float(step['instrument:observingWavelength'])
+        config['wavelength'].append(wavelength)
+
+        p = 0.0
+        q = 0.0
+        # Just want exposures on sky for dither pattern analysis
+        if step['observe:observeType'].upper() not in ['FLAT', 'ARC', 'DARK', 'BIAS']:
+            if 'telescope:p' in step_keys:
+                p = float(step['telescope:p'])
+            else:
+                p = 0.0
+            poffsets_sky.append(p)
+
+            if 'telescope:q' in step_keys:
+                q = float(step['telescope:q'])
+            else:
+                q = 0.0
+            qoffsets_sky.append(q)
+
+        if 'observe:coadds' in step_keys:
+            ncoadds = int(step['observe:coadds'])
+        else:
+            ncoadds = 1
+        poffsets.append(p)
+        qoffsets.append(q)
+
+        exptimes.append(float(step['observe:exposureTime']))
+        coadds.append(ncoadds)
+
+    mode = obsmode(config)
+    if verbose:
+        print(config)
+        print(mode)
+
+    if config['inst'] == 'GPI':
+        do_not_split = True
+
+    # Analyze sky offset patterns using auto-correlation
+    # The lag is the length of any pattern, 0 means no repeating pattern
+    plag = 0
+    qlag = 0
+    offset_lag = 0
+    if do_not_split:
```

Comment:
...and then just change this to:
```
if config['inst'] == 'GPI':
```
I don't see other uses of the `do_not_split` variable so unless some will be added, it seems unnecessary, and this contains the logic to a smaller area, making it more comprehensible.

Reply:
We will need do_not_split in general. For now it is just used for GPI but we may find other cases where it will be used.

Reply:
I'll mark this as resolved, then, with that understanding.